### PR TITLE
doc: provisioning and config info in mesh samples

### PR DESCRIPTION
**Pull request by mia-ko**
NCSDK-7423: Creating template for provisioning doc. Cleaning up model
config.

Signed-off-by: Mia Koen <mia.koen@nordicsemi.no>

### CODE REVIEW

File: doc/nrf/ug_bt_mesh_model_config_app.rst [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+.. contents::
+   :local:
+   :depth: 2
+
+The Bluetooth mesh :ref:`samples <samples>` use the nRF Mesh mobile app to perform provisioning and configuration.
+
+Binding a model to an application key
+*************************************
+
+Complete the following steps in the nRF Mesh app to bind a model to an application key:
```
Comment (trond-snekvik) [inline doc/nrf/ug_bt_mesh_model_config_app.rst:15]: I'd like to add an extra sentence of context to each of these sections. Suggestion:

> Mesh models encrypt all their messages using application keys. Models that should communicate with each other must be bound to the same application key.

File: doc/nrf/ug_bt_mesh_model_config_app.rst [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+   a. Tap :guilabel:`BIND KEY` at the top of the screen.
+   #. Select the key from the list.
+
+#. Tap :guilabel:`APPLY`, in the right bottom corner of the screen, to confirm the configuration.
+
+Configuring model publication parameters
+****************************************
+
+To configure the model publication parameters, which define how the model will send its messages, complete the following steps in the nRF Mesh app:
```
Comment (trond-snekvik) [inline doc/nrf/ug_bt_mesh_model_config_app.rst:32]:
```suggestion
The publication parameters define how model sends its messages.
To configure the model publication parameters, complete the following steps in the nRF Mesh app:
```

File: doc/nrf/ug_bt_mesh_model_config_app.rst [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+   a. Scroll down to the :guilabel:`Publish Period` view and set the publication interval by using the :guilabel:`Interval` slider.
+      This will make the node publish its presence status periodically at the defined interval.
+   b. Change the publication retransmission configuration under the :guilabel:`Publish Retransmission` section. Set the value for the Retransmit Count.
+
+#. Tap :guilabel:`APPLY`, in the right bottom corner of the screen, to confirm the configuration.
+
+Subscribing to a group
+**********************
+
+To configure subscription parameters, complete the following steps in the nRF Mesh mobile app:
```
Comment (trond-snekvik) [inline doc/nrf/ug_bt_mesh_model_config_app.rst:51]: Suggestion for context sentence:

> To receive messages sent to group addresses, models must subscribe to them.

File: doc/nrf/ug_bt_mesh_model_config_app.rst [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+1. Tap :guilabel:`SUBSCRIBE` under the :guilabel:`Subscriptions` section.
+#. Select :guilabel:`Groups`.
+#. Select an existing group or create a new one.
+#. Tap :guilabel:`OK`.
+#. Double-tap the back arrow button at the top left corner of the app to get back to the main application screen.
+
+Creating a group
+****************
+
+Complete the following steps in the nRF Mesh mobile app to create a new group:
```
Comment (trond-snekvik) [inline doc/nrf/ug_bt_mesh_model_config_app.rst:62]: Suggestion:
> To send to more than one node, we must define groups for the models to publish and subscribe to.

Also, perhaps this should go above "Subscribing to a group"?

**Review by trond-snekvik — CHANGES_REQUESTED** [review]
Looks a lot cleaner! I have a couple of minor comments

File: doc/nrf/ug_bt_mesh_model_config_app.rst (hunk 4 above)
Comment (mia-ko) [reply doc/nrf/ug_bt_mesh_model_config_app.rst:62]: Yup, I agree actually. I was a bit back and forth on this one though, but I will move it up.

File: doc/nrf/ug_bt_mesh_model_config_app.rst [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+
+1. On the Groups screen, tap :guilabel:`CREATE GROUP`.
+#. Ensure that the group address type is selected from the drop-down menu.
+#. In the :guilabel:`Name` field, enter the group name.
+#. Enter the group address in the :guilabel:`Address` field.
```
Comment (peknis) [inline doc/nrf/ug_bt_mesh_model_config_app.rst:67]: A minor point about steps 3 and 4:

You start step 3: In the...
But step 4: Enter...

For the consistency, you might want to start step 3 the same way as step 4.

File: samples/bluetooth/mesh/light/README.rst [hunk 6]
```diff
@@ -98,35 +98,23 @@ Testing consists of provisioning the device and configuring it for communication
...
-#. Bind the model to application keys to make it open for communication:
+Configure the Generic OnOff Client model and the parameters on each element for the :guilabel:`Mesh Light` node:
 
-   1. Tap :guilabel:`BIND KEY` at the top of the screen.
-   #. Select :guilabel:`Application Key 1` from the list.
+* Bind the model to :guilabel:`Application Key 1`.
 
-   You are now able to control the first LED on the device by using the Generic On Off Controls in the model view.
-#. Tap :guilabel:`ON` to light up the first LED on the development kit.
+  You are now able to control the first LED on the device by using the Generic On Off Controls in the model view.
```
[the remark this answers is not in the document]
Comment (mia-ko) [reply samples/bluetooth/mesh/light/README.rst:114]: It seems like it "belongs" to the binding the app key bullet point. As some sort of the step result. And as such, it looks ok that it is indented in my opinion. Rather that than continuing in the same line as a new sentence. What do you think?

[the remark this answers is not in the document]
Comment (trond-snekvik) [reply samples/bluetooth/mesh/light/README.rst:114]: I was just confused when looking at the next bullet point, as it doesn't let you know where to find the ON button. I wasn't expecting the sentence of the previous bullet point to contain this information. I don't think it matters that much though

File: samples/bluetooth/mesh/light/README.rst [hunk 7]
```diff
@@ -98,35 +98,23 @@ Testing consists of provisioning the device and configuring it for communication
...
+* Tap :guilabel:`ON` to light up the first LED on the development kit.
 
-Repeat steps 3-5 for each of the elements on the node to enable controling each of the remaining three LEDs.
+Make sure to complete the configuration on each of the elements on the node to enable controling each of the remaining three LEDs.
```
Comment (peknis) [inline samples/bluetooth/mesh/light/README.rst:118]: I believe controlling needs two L:s

File: samples/bluetooth/mesh/light_ctrl/README.rst [hunk 8]
```diff
@@ -120,33 +120,21 @@ Testing consists of provisioning the device and configuring it for communication
...
-#. Bind the model to application keys to make it open for communication:
+Configure the Generic OnOff Client model on each element on the :guilabel:`Mesh Light Fixture` node:
 
-   1. Tap :guilabel:`BIND KEY` at the top of the screen.
-   #. Select :guilabel:`Application Key 1` from the list.
+* Bind the model to :guilabel:`Application Key 1`.
 
-   You are now able to control the first LED on the device by using the Generic On Off Controls in the model view.
-#. Tap :guilabel:`ON` to light up the first LED on the development kit.
+  You are now able to control the first LED on the device by using the Generic On Off Controls in the model view.
```
Comment (peknis) [inline samples/bluetooth/mesh/light_ctrl/README.rst:136]: You can now...

File: samples/bluetooth/mesh/sensor_client/README.rst [hunk 9]
```diff
@@ -114,56 +114,24 @@ For more details, see :ref:`gs_testing`.
...
-         The Sensor Server must publish to the same group.
+* Bind the model to :guilabel:`Application Key 1`.
+* Set the publication parameters:
 
-   #. Tap :guilabel:`OK`.
+  * Destination/publish address: Select an existing group or create a new one, but make sure that the Sensor Server subscribes to that same group.
+  * Retransmit count: Zero (:guilabel:`Disabled`), to avoid duplicate logging in the UART terminal.
 
-#. Double-tap the back arrow button at the top left corner of the app to get back to the main application screen.
+* Set the subscription parameters: Select an existing group or create a new one, but make sure that the Sensor Server publishes to that same group.
```
Comment (peknis) [inline samples/bluetooth/mesh/sensor_client/README.rst:135]: to that same group-> to the same group

File: samples/bluetooth/mesh/sensor_server/README.rst [hunk 10]
```diff
@@ -112,59 +112,26 @@ Testing consists of provisioning the device, and configuring it for communicatio
...
-   #. Select an existing group or create a new one.
+Configure the Sensor Server model on the :guilabel:`Mesh Sensor` node:
 
-      .. note::
-         The Sensor Client must publish to the same group.
+* Bind the model to :guilabel:`Application Key 1`.
+* Set the publication parameters:
 
-   #. Tap :guilabel:`OK`.
+  * Destination/publish address: Select an existing group or create a new one, but make sure that the Sensor Client subscribes to that same group.
```
Comment (peknis) [inline samples/bluetooth/mesh/sensor_server/README.rst:131]: to that same group -> to the same group

File: samples/bluetooth/mesh/sensor_server/README.rst [hunk 11]
```diff
@@ -112,59 +112,26 @@ Testing consists of provisioning the device, and configuring it for communicatio
...
+  * Retransmit count: Zero (:guilabel:`Disabled`), to avoid duplicate logging in the :ref:`bt_mesh_sensor_cli_readme`'s UART terminal.
 
-#. Double-tap the back arrow button at the top left corner of the app to get back to the main application screen.
+* Set the subscription parameters: Select an existing group or create a new one, but make sure that the Sensor Client publishes to that same group.
```
Comment (peknis) [inline samples/bluetooth/mesh/sensor_server/README.rst:134]: to that same group -> to the same group

File: doc/nrf/ug_bt_mesh_model_config_app.rst [hunk 12]
```diff
@@ -0,0 +1,74 @@
...
+#. Select an existing group or create a new one.
+#. Tap :guilabel:`OK`.
+#. If necessary, set the publication interval and publication retransmission configuration.
+   Otherwise, leave the publication parameters at their default values.
+
+   a. Scroll down to the :guilabel:`Publish Period` view and set the publication interval by using the :guilabel:`Interval` slider.
+      This will make the node publish its presence status periodically at the defined interval.
+   b. Change the publication retransmission configuration under the :guilabel:`Publish Retransmission` section. Set the value for the Retransmit Count.
+
+#. Tap :guilabel:`APPLY`, in the right bottom corner of the screen, to confirm the configuration.
```
Comment (peknis) [inline doc/nrf/ug_bt_mesh_model_config_app.rst:62]: Dunno whether the commas are needed here.

File: doc/nrf/ug_bt_mesh_model_config_app.rst [hunk 13]
```diff
@@ -0,0 +1,74 @@
...
+
+   a. Tap :guilabel:`BIND KEY` at the top of the screen.
+   #. Select the key from the list.
+
+#. Tap :guilabel:`APPLY`, in the right bottom corner of the screen, to confirm the configuration.
+
+Creating a group
+****************
+
+To send messages to more than one node in the mesh network, we must define groups for the models to publish and subscribe to.
```
Comment (peknis) [inline doc/nrf/ug_bt_mesh_model_config_app.rst:35]: we -> you

**Review by peknis — CHANGES_REQUESTED** [review]

File: samples/bluetooth/mesh/chat/README.rst [hunk 14]
```diff
@@ -103,57 +107,27 @@ After configuring the device, you can interact with the sample using the termina
...
-      This will make the node publish its presence status periodically at the defined interval. Recommended value is 10 seconds.
-   #. You can also change the publication retransmission configuration under the :guilabel:`Publish Retransmission` section.
-   #. Tap :guilabel:`APPLY` button in the right bottom corner of the screen to apply the changes.
+* Bind the model to :guilabel:`Application Key 1`.
+* Set the publication parameters:
 
-#. Set a model subscription:
+  * Destination/publish address: Select just created group :guilabel:`Chat Channel`.
+  * Publication interval: Recommended value is 10 seconds.
+  * Retransmit count: Can change it if desired.
```
[the remark this answers is not in the document]
Comment (mia-ko) [reply samples/bluetooth/mesh/chat/README.rst]: Changed it to: Change the count as preferred.
And changed the second bullet point to: Set the interval to recommended value of 10 seconds.

For consistency :)

File: samples/bluetooth/mesh/light/README.rst (hunk 6 above)
[the remark this answers is not in the document]
Comment (mia-ko) [reply samples/bluetooth/mesh/light/README.rst]: I shuffled it around a bit. See if you think it's better :)

File: samples/bluetooth/mesh/light/README.rst (hunk 7 above)
Comment (mia-ko) [reply samples/bluetooth/mesh/light/README.rst:118]: hahaha You are right, this one I totally went blind on :)

File: doc/nrf/ug_bt_mesh_model_config_app.rst (hunk 12 above)
Comment (mia-ko) [reply doc/nrf/ug_bt_mesh_model_config_app.rst:62]: Tja... not incorrect, but we might not needed. I can remove it while I'm at it.

**Review by peknis — APPROVED** [review]

File: samples/bluetooth/mesh/light/README.rst (hunk 6 above)
[the remark this answers is not in the document]
Comment (trond-snekvik) [reply samples/bluetooth/mesh/light/README.rst]: Looks good!

**Review by trond-snekvik — APPROVED** [review]

**Pull request merged**